plugin/git/commands.go

```diff
@@ -0,0 +1,48 @@
+package git
+
+import (
+	"bytes"
+	"os/exec"
```

Comment:
I'm a big un-fan of having `os/exec` in a service like this. Having CoreDNS fork processes in the background seems like a bad idea.

Most people who wan to store config in git will have some kind of external management like Ansible/Chef/Kubernetes.

Reply:
so don't use this plugin?
I don't see why executing a git command every hour would be a bad thing? Can you explain?

Reply:
Because these things are better handled async, outside of a serving process. For example, a cron job.

Counter question, what is the advantage of re-inventing cron here?